plugin/pkg/admission/priority/admission.go

```diff
@@ -196,8 +196,14 @@ func (p *priorityPlugin) admitPod(a admission.Attributes) error {
 			pod.Spec.PriorityClassName = pcName
 		} else {
 			pcName := pod.Spec.PriorityClassName
-			if !priorityClassPermittedInNamespace(pcName, a.GetNamespace()) {
-				return admission.NewForbidden(a, fmt.Errorf("pods with %v priorityClass is not permitted in %v namespace", pcName, a.GetNamespace()))
+			// If ResourceQuotaScopeSelectors is enabled, we should let critical priorityClass to be created
+			// any namespace where administrator wants it to be created.
+			// TODO: @ravig Add a default quota at cluster bootstrap which limits critical priorityClasses
+			// to be in kube-system namespace. This can be modified later.
+			if !utilfeature.DefaultFeatureGate.Enabled(features.ResourceQuotaScopeSelectors) {
```

Comment:
Our recent benchmarks show that checking feature gates is relatively slow. We should avoid checking feature gates here. Instead we should check the feature gates at the time of plugin creation and store it in a member variable and use that variable instead.